### PR TITLE
Enhance cursor offset saving/restoring

### PR DESCRIPTION
**Pull request by Ind1eMonk3y**

### CODE REVIEW

File: lua/harpoon/mark.lua [hunk 1]
```diff
@@ -144,6 +144,7 @@ M.add_file = function(file_name_or_buf_id)
 
     local found_idx = get_first_empty_slot()
     harpoon.get_mark_config().marks[found_idx] = create_mark(buf_name)
+    M.store_offset()
```
Comment (ThePrimeagen) [inline lua/harpoon/mark.lua:147]: THis is the wrong place.  add_file calls create mark which creates a mark with the offset already!

Comment (ThePrimeagen) [reply lua/harpoon/mark.lua:147]: You should be able to remove it here, or update create_mark to make sure it puts the rows and cols in.

**Comment by Ind1eMonk3y** [comment]
I hope it's fine now.

**Comment by ThePrimeagen** [comment]
I know you went to bed, but when you wake up squashyour commits into one and we got ourselves a pr!  :)

**Comment by disrupted** [comment]
FYI you can also squash when merging

**Pull request merged**